CMakeLists.txt

```diff
@@ -486,8 +487,12 @@ elseif(MSVC)
 endif()
 
 if(static_runtime)
-	include(ucm_flags)
-	ucm_set_runtime(STATIC)
+	if (MSVC)
+		set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
```

Comment:
so, cmake just requires you to set this variable when building static libraries with msvc?
does it not affect MingW? (it's not obvious that `MSVC` is the correct test).

is `ucm_set_runtime()` still required on other compilers? (still, it seems like it's more of a target platform distinction, rather than compiler). It seems a bit strange that cmake couldn't figure out which runtime library to link against on its own.

I think these things are a bit mysterious, at least to someone who isn't a cmake expert. A comment could explain some of this, or maybe provide a link to an explanation of what this does.

What runtime library does cmake link against if this is *not* set?

Reply:
>so, cmake just requires you to set this variable when building static libraries with msvc?

It appears so.

>does it not affect MingW? (it's not obvious that MSVC is the correct test).

The variable `CMAKE_MSVC_RUNTIME_LIBRARY` is explicitly for MSVC as the name is in it.

>What runtime library does cmake link against if this is not set?

See the link in the commit, I'll quote it for you:
> If they are not set then CMake uses the default value MultiThreaded$<$<CONFIG:Debug>:Debug>DLL which is equivalent to the original flags.

>is ucm_set_runtime() still required on other compilers? 

Seems yes, but I don't build statically on other platforms. This PR only aim to fix compilation with MSVC.

Reply:
> The variable CMAKE_MSVC_RUNTIME_LIBRARY is explicitly for MSVC as the name is in it.

good point